### PR TITLE
Mutex Locks

### PR DESCRIPTION
**Pull request by pvelesko**
* Remove all calls to std::mutex Lock from backend implementations
* All locks now done at CHIP-SPV level to prevent deadlocks

### CODE REVIEW

File: src/CHIPBackend.cc [hunk 1]
```diff
@@ -1354,6 +1363,7 @@ void CHIPQueue::memCopyToTexture(CHIPTexture *texObj, void *src) {
   updateLastEvent(ev);
 }
 CHIPEvent *CHIPQueue::launch(CHIPExecItem *exec_item) {
+  // std::lock_guard<std::mutex> Lock(mtx);
```
Comment (linehill) [inline src/CHIPBackend.cc:1366]: Is this commented on purpose?

File: src/backend/OpenCL/CHIPBackendOpenCL.cc [hunk 2]
```diff
@@ -250,7 +249,7 @@ bool CHIPEventOpenCL::updateFinishStatus() {
 float CHIPEventOpenCL::getElapsedTime(CHIPEvent *other_) {
   // Why do I need to lock the context mutex?
   // Can I lock the mutex of this and the other event?
-  // std::lock_guard<std::mutex> Lock(mtx);
+  //
```
Comment (linehill) [inline src/backend/OpenCL/CHIPBackendOpenCL.cc:252]: Should remove empty comment line?

File: src/backend/OpenCL/CHIPBackendOpenCL.cc [hunk 3]
```diff
@@ -491,7 +459,7 @@ CHIPEventOpenCL *CHIPQueueOpenCL::getLastEvent() {
 }
 
 CHIPEvent *CHIPQueueOpenCL::launchImpl(CHIPExecItem *exec_item) {
-  // std::lock_guard<std::mutex> Lock(mtx);
+  //
```
Comment (linehill) [inline src/backend/OpenCL/CHIPBackendOpenCL.cc:462]: Empty comment line here too.

File: src/CHIPBindings.cc [hunk 4]
```diff
@@ -719,7 +719,9 @@ hipError_t hipStreamWaitEvent(hipStream_t stream, hipEvent_t event,
   ERROR_IF((stream == nullptr), hipErrorInvalidResourceHandle);
   ERROR_IF((event == nullptr), hipErrorInvalidResourceHandle);
 
-  event->barrier(stream);
+  std::vector<CHIPEvent *> EventsToWaitOn = {event};
+  stream->enqueueBarrier(&EventsToWaitOn);
+  // event->barrier(stream);
```
Comment (linehill) [inline src/CHIPBindings.cc:724]: Should this comment removed?

**Comment by pvelesko** [comment]
This is draft PR, please disregard for the time being

**Comment by linehill** [comment]
I see. Didn't noticed the draft label.

**Pull request closed without merging**